### PR TITLE
Megafauna cannot teleport

### PR DESCRIPTION
**Pull request by 4dplanner**
:cl: 
balance: Megafauna cannot teleport
/:cl:

It's easy to do and destroys the entire station, especially with the reworked bosses. Total mismatch of risk and reward.

https://youtu.be/zsC1JNxGMNc

### CODE REVIEW

**Comment by 4dplanner** [comment]
When cyberboss says that he's going to be super unavailable but still finds the time to put shitty labels on your prs

**Comment by Ziiro** [comment]
another ~~fine~~ i ded PR from 4dplanner

~~I guess you have plenty of time to grudge code with how much time you spend dead?~~

"Total mismatch of risk and reward." You risk permadeath via engaging a megafauna to destroy the station. Destabilizing the SM has less risk. Building a toxins bomb has less risk. Plus all it takes is another big brain to throw up another teleporter and send it to the abandoned teleporter to get it to leave.

**Comment by 4dplanner** [comment]
I know, I'll tell him he ded! That'll convince him!

**Comment by Daxxed** [comment]
Are you saying theres no risk in using a hierophant staff ?
You need to focus more on the specific issue with your PRs instead of nerfing entire departments because of one overpowered item.

**Comment by 4dplanner** [comment]
Hierophant staff is not a teleport, it's a forcemove, and is unaffected by this pr.

**Comment by Daxxed** [comment]
And since you dont even have to go back to station you could do it all while having the comfort of unpressurized space where PKA is king and eventually greentext on the white ship with no risk involved at all.

**Comment by Daxxed** [comment]
> 
> 
> Lame as fuck tbh

Yes indeed using a hand tele is lame as fuck.
Real men stand in front of colossus charging the hieroport for 5 seconds.

**Comment by Swiftkor** [comment]
You can still lure them on shuttles, correct? If so then I’ll change to an upvote, I thought you were completely removing bringing megafauna to the station.

**Comment by 4dplanner** [comment]
Here is footage of me risking my life to earn a great reward: https://youtu.be/zsC1JNxGMNc

**Comment by 4dplanner** [comment]
Yeah, shuttle should be unaffected.

File: code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm [hunk 1]
```diff
@@ -38,6 +38,7 @@
 /mob/living/simple_animal/hostile/megafauna/Initialize(mapload)
 	. = ..()
 	apply_status_effect(STATUS_EFFECT_CRUSHERDAMAGETRACKING)
+	add_trait(TRAIT_NO_TELEPORT, MEGAFAUNA_TRAIT)
```
Comment (Trilbyspaceclone) [inline code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm]: Dose this affect BDM or Bubblegum?

Comment (4dplanner) [reply code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm]: Should affect all of them by inheritance, admittedly only tested with bubblegum and drake

Comment (Trilbyspaceclone) [reply code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm]: Ok, so dose it mess with there teleports? Or will they be fine, IIRC blood drunk miner dose a new/loc call and I think Bubble gum is able do a teleport move at you when crushing, unless im wrong*

Comment (4dplanner) [reply code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm]: Those special attacks don't use teleport on bubblegum, I'll just check on miner.

Comment (4dplanner) [reply code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm]: Yeah, neither should be affected.

Comment (Trilbyspaceclone) [reply code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm]: Noted.

**Comment by subject217** [comment]
Jaunters working on megafauna was removed for the same reason.

**Comment by wesoda25** [comment]
Lmao at the stupid cunts against this PR. You guys clearly haven’t seen fauna since the update.

**Comment by 4dplanner** [comment]
> Jaunters working on megafauna was removed for the same reason.

When? They worked pretty recently

**Comment by 4dplanner** [comment]
I actually saw a non-antag bubblegum on station when a newbie was splatted

**Comment by subject217** [comment]
https://github.com/tgstation/tgstation/pull/34164

**Comment by Hathkar** [comment]
Xenobio love potion + the teleport crossbreed extract = Fauna on station that is friendly to you and hostile to everyone else.

File: code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm (hunk 1 above)
Comment (Whoneedspacee) [reply code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm]: Yeah this is checking if the portal will teleport after something has passed onto it. It shouldn’t affect forcemove in every way (I should know, my new pr does a ton of new teleporter shit)

**Comment by imsxz** [comment]
@Ziiro you risk nothing when you do anything other than dragging them onto the shuttle, using the shuttle to lure megafauna should be the only viable way to go about it.

As someone who uses methods that this PR would remove, nothing of value is being lost. Megafauna by no means need to be easy to get on station.

**Comment by Ziiro** [comment]
You know what, fair. 👍

**Comment by OverlyGeneric** [comment]
Space ruin megafauna when?

**Comment by Shadowflame909** [comment]
After this PR gets merged, its time to make bubblegum a username heat-seeking missile that retraces his steps after losing sight of someone for a while.


No more bullfighting him into an endless loop in the void of space!

**Comment by Whoneedspacee** [comment]
> After this PR gets merged, its time to make bubblegum a username heat-seeking missile that retraces his steps after losing sight of someone for a while.
> 
> No more bullfighting him into an endless loop in the void of space!

You already can't do that I'm pretty sure, he no longer charges by throwing himself so he can spacewalk normally now. I did test it.

**Comment by Shadowflame909** [comment]
Someone did it.


I know because I brought new bubblegum onto the station like a week ago

**Comment by 4dplanner** [comment]
> After this PR gets merged, its time to make bubblegum a username heat-seeking missile that retraces his steps after losing sight of someone for a while.
> 
> No more bullfighting him into an endless loop in the void of space!

lol no, think of the poor miners

File: code/game/objects/effects/portals.dm [hunk 2]
```diff
@@ -149,7 +149,7 @@
 		return ..()
 
 /obj/effect/portal/proc/teleport(atom/movable/M, force = FALSE)
-	if(!force && (!istype(M) || iseffect(M) || (ismecha(M) && !mech_sized) || (!isobj(M) && !ismob(M)))) //Things that shouldn't teleport.
+	if(!force && (M.has_trait(TRAIT_NO_TELEPORT) || !istype(M) || iseffect(M) || (ismecha(M) && !mech_sized) || (!isobj(M) && !ismob(M)))) //Things that shouldn't teleport.
```
Comment (Cyberboss) [inline code/game/objects/effects/portals.dm]:
```suggestion
	if(!force && (!istype(M) || iseffect(M) || (ismecha(M) && !mech_sized) || (!isobj(M) && !ismob(M)))) //Things that shouldn't teleport.
```

You already checked this in `do_teleport`

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by wesoda25** [comment]
Rip

**Comment by Shadowflame909** [comment]
The worst part is, this PR made people aware of what the hand teleporter could truly do.

**Comment by wesoda25** [comment]
Yeah I bet imsxz is beating themself up for all the times they risked their life to get fauna onto the station. 

Who knew theres a no risk way of killing an entire station?

**Comment by Shadowflame909** [comment]
True that. I had no idea this was a thing, I feel like it was the same for the majority

**Comment by Daxxed** [comment]
Can someone revive this. Risk reward etc.
Real men dont need this to summon megafauna and bubbles has no business on station.

**Comment by Shadowflame909** [comment]
I got a couple questions about this PR B.

Does it stop thy admin teleporto?

What about bubblegums blood-crawl technique?

**Comment by subject217** [comment]
if 4dplanner doesn't feel like this right now i'm gonna try to yeet this towards finished

**Comment by Whoneedspacee** [comment]
@subject217

teleport already has a ismegafauna check so just change that instead of going through all this bullshit

if you want to go one step further make it so megafauna check if they arent on their spawned z level and give them destructive animations to return that’d be funny

*ash drake smashes through the ceiling*

*colossus yells at everyone to stop stunning everyone and teleports back* 

i dont fuckin know thatd be neato though

**Comment by 4dplanner** [comment]
I realise this is actually only one line away from finished so I might just finish it anyway lol

**Comment by Kromgar** [comment]
Oh fuck no. Don't make it so megafauna can't ever be brought to the station. That's fucking lame. Taking them on the shuttle is still highly risk and can result in death so many things can go wrong

**Review by Cyberboss — APPROVED** [review]

**Comment by subject217** [comment]
use the shuttle you chicken shit

nobody *deserves* to bring the almighty death mob to the station with zero risk involved

**Pull request merged**